### ISSUE TITLE
Advice: Regression with PointNet

### ISSUE THREAD

**Issue by shrutirao94** [fxia22/pointnet.pytorch#77]
Hello, 

I am interested in furthering PointNet and using it for regression. Is this possible? Has anyone attempted this yet? Any thoughts or comments would be great!

Thanks!

**Comment by FandosA** [comment]
> Hello,
> 
> I am interested in furthering PointNet and using it for regression. Is this possible? Has anyone attempted this yet? Any thoughts or comments would be great!
> 
> Thanks!

Hello @shrutirao94, I tried it for regression. In my problem, I'm working on a deep neural network able to predict the saliency of a 3D shape (saliency are the parts of a shape that most capture human attention). I have pointclouds as training data (as here), and my labels are float values (instead of discrete labels), corresponding to the time that each point have been fixated in the experiment. The only changes I made were the output size, which now in my case is 1 (one only value per point), and the activation of the final layer, because in this case, as it is a classification problem, they use the log_softmax, and it wouldn't make sense in a regression problem. So I did this change:

```
x = x.transpose(2, 1).contiguous()
x = F.relu(x.view(-1, self.output_size))
        
#x = F.log_softmax(x.view(-1,self.k), dim=-1)  # Esto estaba en el original
#x = x.view(-1, self.output_size)
        
x = x.view(batchsize, n_pts, self.output_size)
```

I hope this is helpful for you. For me, it doesn't work as I would like...

**Comment by shrutirao94** [comment]
Hey @FandosA, thank you for sharing this, reminded me that I should have updated and closed this issue a while back! 
I too ended up making the standard changes such as output layer etc. In my case, I was trying to infer energy values of neutrino particles, but I could not suitable interpret the values the model returned for my problem!

**Comment by FandosA** [comment]
I'm glad this was helpful! Another change that I forgot to mention was the loss function at the ``main``. In this project they use the negative log likelihood because of the same reason, it's a classification problem. So, for a regression problem, this does not make any sense, thus you should change it and use some other loss function such as mean squared error, mean absolute error or the Huber loss.
